# Lab 4 Estymacja


## Zadanie 1

4. Estymacja 1 UWAGA! Większość danych pojawiających się w poniższych zadaniach dostępna jest w pliku dane_est.csv 1. W celu oceny nowego procesu produkcji syntetycznych diamentów sprawdzono wagę [karaty] diamentów wyprodukowanych tą metodą uzyskując następujące wyniki: 0,46 0,61 0,52 0,48 0,57 0,54 0,47 0,63 0,51 0,49 0,58 0,55. Przyjmijmy, że badana zmienna ma rozkład normalny. (a) Określ populację, próbę i badaną zmienną. (b) Wyznacz ocenę punktową średniej wagi diamentów produkowanych tą metodą. 0,534167 (c) Oszacuj z 95% pewnością średnią wagę wszystkich syntetycznych diamentów produkowanych badaną metodą (skonstruuj własną funkcję i porównaj wynik z wynikami odpowiedniej funkcji dostępnej w R). (0,498; 0,57) (d) Zwiększ ufność z jaką chcemy wnioskować i porównaj długości uzyskanych przedziałów ufności.


In [3]:
# (a) Określenie populacji, próby i badanej zmiennej
# Wczytanie danych z pliku CSV



dane <- read.csv("dane/dane_est.csv", sep = ";", dec = ",", header = TRUE)

# (b) Wyznacz ocenę punktową średniej wagi diamentów
# Pobranie danych o diamentach (pomijając NA)
wagi <- na.omit(dane$diamenty)
srednia <- mean(wagi)
print(srednia)

# (c) Oszacuj średnią z 95% przedziałem ufności
# Własna funkcja
przedzial_ufnosci <- function(x, poziom = 0.95) {
  n <- length(x)
  s <- sd(x)
  m <- mean(x)
  se <- s / sqrt(n)
  t <- qt(1 - (1 - poziom) / 2, df = n - 1)
  dol <- m - t * se
  gora <- m + t * se
  c(dol, gora)
}

# Wywołanie własnej funkcji
pi_95 <- przedzial_ufnosci(wagi)
print(pi_95)

# Wbudowana funkcja
pi_95_t <- t.test(wagi, conf.level = 0.95)$conf.int
print(pi_95_t)

# (d) Przedział ufności 99% i porównanie długości
# Zwiększenie poziomu ufności do 99%
pi_99 <- przedzial_ufnosci(wagi, poziom = 0.99)
print(pi_99)

pi_99_t <- t.test(wagi, conf.level = 0.99)$conf.int
print(pi_99_t)

# Porównanie długości przedziałów
dlugosc_95 <- pi_95[2] - pi_95[1]
dlugosc_99 <- pi_99[2] - pi_99[1]
cat("Długość przedziału 95%:", dlugosc_95, "\n")
cat("Długość przedziału 99%:", dlugosc_99, "\n")
cat("Stosunek długości 99% do 95%:", dlugosc_99 / dlugosc_95)

[1] 0.5341667
[1] 0.4988990 0.5694343
[1] 0.4988990 0.5694343
attr(,"conf.level")
[1] 0.95
[1] 0.4844005 0.5839328
[1] 0.4844005 0.5839328
attr(,"conf.level")
[1] 0.99


Długość przedziału 95%: 0.07053531 
Długość przedziału 99%: 0.09953227 
Stosunek długości 99% do 95%: 1.411098

### Metoda

#### (a)

Określenie populacji, próby i badanej zmiennej jest zadaniem koncepcyjnym, nie wymaga analizy statystycznej.

#### (b)

Średnia arytmetyczna to estymator punktowy wartości oczekiwanej (średniej) w populacji. Przy założeniu rozkładu normalnego jest to estymator nieobciążony i zgodny.

#### (c)

Przy konstrukcji przedziału ufności dla średniej z rozkładu normalnego o nieznanej wariancji korzystamy z rozkładu t-Studenta. Przedział ufności na poziomie 1-α ma postać:
`średnia ± t(α/2, n-1) * (s / √n)`,
gdzie t(α/2, n-1) to kwantyl rzędu 1-α/2 z rozkładu t-Studenta z n-1 stopniami swobody, s to odchylenie standardowe z próby, a n to liczebność próby.

#### (d)

Zwiększając poziom ufności (np. z 95% do 99%), zwiększamy wartość kwantyla t, co prowadzi do poszerzenia przedziału ufności.

### Opis

#### (a)

Określamy populację jako wszystkie syntetyczne diamenty, które mogą być wyprodukowane badaną metodą. Próba składa się z 12 diamentów, których wagi zostały zmierzone i zapisane w pliku CSV w kolumnie "diamenty". Badaną zmienną jest waga diamentów mierzona w karatach.

#### (b)

Wczytujemy dane z pliku CSV przy użyciu funkcji `read.csv()` z odpowiednimi parametrami:

- `sep = ";"` - określa separator kolumn jako średnik
- `dec = ","` - określa, że przecinek jest używany jako separator dziesiętny
- `header = TRUE` - określa, że pierwszy wiersz zawiera nazwy kolumn

Następnie pobieramy kolumnę "diamenty" i usuwamy ewentualne wartości NA przy użyciu funkcji `na.omit()`. Obliczamy średnią arytmetyczną z próby 12 diamentów przy użyciu funkcji `mean()`.

#### (c)

Tworzymy funkcję `przedzial_ufnosci()`, która:

1. Wyznacza liczebność próby, odchylenie standardowe i średnią
2. Oblicza błąd standardowy średniej (s / √n)
3. Znajduje odpowiedni kwantyl z rozkładu t-Studenta
4. Oblicza dolną i górną granicę przedziału ufności
5. Zwraca przedział jako wektor dwuelementowy

Porównujemy wynik z wbudowaną funkcją `t.test()`, która wykonuje te same obliczenia.

#### (d)

Zwiększamy poziom ufności z 95% do 99%, obliczamy oba przedziały i porównujemy ich długości. Długość przedziału to różnica między górną a dolną granicą. Obliczamy również stosunek długości, aby zobaczyć, o ile szerszy jest przedział o wyższej ufności.

### Interpretacja

#### (a)

Populacja to wszystkie możliwe diamenty wyprodukowane badaną metodą produkcji syntetycznych diamentów. Próba to 12 zmierzonych diamentów, a badana zmienna to waga diamentów mierzona w karatach.

#### (b)

Średnia waga diamentów w próbie wynosi około 0,534 karata. Jest to nasze najlepsze punktowe oszacowanie średniej wagi wszystkich diamentów, które mogą być wyprodukowane tą metodą.

#### (c)

Z 95% pewnością możemy stwierdzić, że prawdziwa średnia waga wszystkich diamentów produkowanych badaną metodą zawiera się w przedziale od około 0,498 do 0,57 karata. Oznacza to, że gdybyśmy wielokrotnie pobierali próby i konstruowali przedziały ufności, to około 95% z nich zawierałoby prawdziwą wartość średniej w populacji.

#### (d)

Przedział ufności 99% jest szerszy niż przedział 95%, co jest oczekiwane, ponieważ większa pewność wymaga większej "ostrożności" w oszacowaniu. Zwiększenie poziomu ufności z 95% do 99% powoduje poszerzenie przedziału - jest to kompromis między precyzją (wąski przedział) a pewnością (wysoki poziom ufności). Im wyższa pewność, tym mniej precyzyjne oszacowanie, co widać po stosunku długości tych przedziałów.


## Zadanie 2

Agencja Ochrony Środowiska jest zaniepokojona ilością PCB – toksycznej substancji chemicznej – w mleku matek karmiących piersią. W próbie 20 kobiet poziom PCB (w liczbie cząsteczek na milion) był następujący: 16, 0, 0, 2, 3, 6, 8, 2, 5, 0, 12, 10, 5, 7, 2 , 3, 8, 17, 9, 1. Załóżmy, że rozkład analizowanej zmiennej losowej jest normalny. (a) Zdefiniuj populację, próbkę i badaną zmienną. (b) Oblicz szacunkowy średni poziom PCB w mleku wszystkich matek karmiących piersią. 5.8 (c) Oceń z ufnością 95% średni poziom PCB w mleku wszystkich matek karmiących piersią. Zinterpretuj wynik. (3,42; 8,18)


In [5]:
# (a) Określenie populacji, próby i zmiennej
# Nie wymaga kodu R

# (b) Obliczenie szacunkowego średniego poziomu PCB
# Najpierw zdefiniujmy dane z poziomem PCB
pcb <- c(16, 0, 0, 2, 3, 6, 8, 2, 5, 0, 12, 10, 5, 7, 2, 3, 8, 17, 9, 1)

# Obliczamy średnią
mean(pcb)

# (c) Przedział ufności 95% dla średniego poziomu PCB
# Możemy obliczyć ręcznie
n <- length(pcb)
s <- sd(pcb)
se <- s/sqrt(n)
t_val <- qt(0.975, df=n-1)
dolna_granica <- mean(pcb) - t_val*se
gorna_granica <- mean(pcb) + t_val*se
c(dolna_granica, gorna_granica)

# Alternatywnie, możemy użyć wbudowanej funkcji t.test
t.test(pcb, conf.level=0.95)$conf.int

# Możemy też wczytać dane z pliku CSV
dane <- read.csv("dane/dane_est.csv", sep=";", dec=",")
pcb_z_pliku <- na.omit(dane$mleko)
mean(pcb_z_pliku)
t.test(pcb_z_pliku, conf.level=0.95)$conf.int

[1] 5.8

[1] 3.420358 8.179642

[1] 3.420358 8.179642
attr(,"conf.level")
[1] 0.95

[1] 5.8

[1] 3.420358 8.179642
attr(,"conf.level")
[1] 0.95

### Metoda

#### (a)

Definiowanie populacji, próby i zmiennej badanej jest elementem analizy statystycznej. Nie wymaga obliczeń.

#### (b)

Średnia arytmetyczna jest estymatorem punktowym średniej populacji. Przy założeniu normalności rozkładu, jest to najlepszy estymator nieobciążony.

#### (c)

Do oszacowania średniego poziomu PCB z określonym poziomem ufności stosujemy przedział ufności oparty na rozkładzie t-Studenta, ponieważ zakładamy rozkład normalny zmiennej, a odchylenie standardowe populacji nie jest znane.

### Opis

#### (a)

Populacja to wszystkie matki karmiące piersią.
Próba to 20 kobiet, których poziom PCB zmierzono.
Badana zmienna to poziom PCB w mleku (liczba cząsteczek na milion).

#### (b)

Obliczamy średnią arytmetyczną z próby 20 kobiet, sumując wszystkie wartości poziomu PCB i dzieląc przez liczbę obserwacji. W kodzie używamy funkcji `mean()` w R.

#### (c)

Konstruujemy przedział ufności na poziomie 95% używając wzoru:
średnia ± t\_(α/2, n-1) \* (s/√n), gdzie:

- średnia to średnia z próby
- t\_(α/2, n-1) to wartość krytyczna z rozkładu t-Studenta dla n-1 stopni swobody i poziomu istotności α/2
- s to odchylenie standardowe z próby
- n to liczebność próby

W R obliczamy to ręcznie w krokach, a dodatkowo weryfikujemy wynik używając wbudowanej funkcji `t.test()`.

### Interpretacja

#### (a)

Badamy poziom PCB w mleku matek karmiących piersią, aby ocenić narażenie populacji matek karmiących na tę toksyczną substancję. PCB (polichlorowane bifenyle) są substancjami szkodliwymi dla zdrowia, dlatego ważne jest monitorowanie ich poziomu.

#### (b)

Średni poziom PCB w mleku badanych 20 matek karmiących piersią wynosi 5,8 cząsteczek na milion. Jest to punkt odniesienia dla dalszych badań i ocen ryzyka.

#### (c)

Z 95% pewnością możemy stwierdzić, że średni poziom PCB w mleku wszystkich matek karmiących piersią zawiera się w przedziale od 3,42 do 8,18 cząsteczek na milion.

Interpretacja: Oznacza to, że gdybyśmy wielokrotnie pobierali próby o wielkości 20 kobiet i konstruowali przedziały ufności, to w 95% przypadków prawdziwa średnia poziomu PCB w populacji wszystkich matek karmiących znalazłaby się w tych przedziałach.

Ponieważ cały przedział zawiera wartości większe od zera, możemy wnioskować, że istnieje problem obecności PCB w mleku matek karmiących. Agencja Ochrony Środowiska powinna rozważyć dalsze badania i ewentualne działania mające na celu zmniejszenie narażenia matek karmiących na PCB.


## Zadanie 3

Aby oszacować średnią zawartość nikotyny w nowej marce papierosów, wybrano 15 paczek papierosów i zbadano w nich zawartość nikotyny otrzymując dane (w mg): 1,87 2,28 1,77 2,13 1,43 1,64 2,38 1,39 1,94 2,68 1,95 0,86 1,98 1,69 1,15. Z wcześniejszych badań wiadomo, że rozkład zawartości nikotyny jest normalny z odchyleniem standardowym równym 0,7 mg. (a) Oceń z ufnością 95% średnią zawartości nikotyny we wszystkich papierosach? (1,455; 2,164) (b) Jak duża próbka jest potrzebna, aby długość 95% przedziału ufności była nie większa niż 0,3 mg? 84

Wczytaj dane z pliku csv dane_est.csv ktory zbudowany jest tak :
diamenty;mleko;papierosy;wodorosty;sygnal


In [8]:
install.packages("BSDA")

Installing package into ‘/home/khal/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)

also installing the dependencies ‘proxy’, ‘e1071’




In [10]:
library(BSDA)

In [11]:
# Wczytywanie danych z pliku CSV
dane <- read.csv("dane/dane_est.csv", sep=";", dec=",", header=TRUE)

# (a) Oszacowanie średniej zawartości nikotyny z ufnością 95%
nikotyna <- na.omit(dane$papierosy)  # Usuwanie NA z danych o nikotynie

wynik_z_test <- z.test(nikotyna, sigma.x=0.7, conf.level=0.95)
przedział_ufnosci <- wynik_z_test$conf.int
print(przedział_ufnosci)

# (b) Wielkość próby dla długości przedziału ufności nie większej niż 0,3 mg
dlugosc_maksymalna <- 0.3
z <- qnorm(0.975)
sigma <- 0.7

wymagana_wielkosc_probki <- ceiling((2 * z * sigma / dlugosc_maksymalna)^2)
print(wymagana_wielkosc_probki)

[1] 1.455091 2.163576
attr(,"conf.level")
[1] 0.95


[1] 84


### Metoda

#### (a)

Do oszacowania średniej zawartości nikotyny używamy przedziału ufności dla średniej przy znanym odchyleniu standardowym (σ = 0,7 mg). Ponieważ znamy odchylenie standardowe populacji i zakładamy rozkład normalny, stosujemy test z (rozkład normalny) z wykorzystaniem funkcji `z.test` z pakietu BSDA. Dane wczytujemy z pliku CSV.

#### (b)

Aby znaleźć wymaganą wielkość próby, korzystamy z wzoru na długość przedziału ufności i rozwiązujemy go dla n.

### Opis

#### (a)

1. Wczytujemy dane z pliku CSV używając funkcji `read.csv()` z odpowiednimi parametrami.
2. Wyodrębniamy kolumnę "papierosy" zawierającą dane o zawartości nikotyny.
3. Usuwamy brakujące wartości (NA) za pomocą funkcji `na.omit()`.
4. Używamy funkcji `z.test` z pakietu BSDA do obliczenia 95% przedziału ufności dla średniej.
5. Funkcja ta wymaga podania wektora danych, znanego odchylenia standardowego populacji (sigma.x=0.7) oraz poziomu ufności (conf.level=0.95).
6. Wyodrębniamy przedział ufności z wyników testu za pomocą `wynik_z_test$conf.int`.

#### (b)

1. Długość przedziału ufności to 2 _ z\_{α/2} _ (σ/√n).
2. Chcemy, aby ta długość była nie większa niż 0,3 mg.
3. Przekształcamy wzór: 2 _ z\_{α/2} _ (σ/√n) ≤ 0,3.
4. Rozwiązujemy dla n: n ≥ (2 _ z\_{α/2} _ σ / 0,3)².
5. Podstawiamy wartości: z\_{α/2} = 1,96 (dla 95% ufności), σ = 0,7.
6. Zaokrąglamy wynik w górę za pomocą funkcji `ceiling()`, ponieważ liczba próbek musi być liczbą całkowitą.

### Interpretacja

#### (a)

Z 95% ufnością można stwierdzić, że prawdziwa średnia zawartość nikotyny we wszystkich papierosach tej marki mieści się w przedziale od około 1,455 mg do 2,164 mg. Oznacza to, że gdybyśmy wielokrotnie pobierali próby i tworzyli przedziały ufności, to 95% tych przedziałów zawierałoby rzeczywistą średnią zawartość nikotyny w populacji. Dane zostały wczytane z pliku CSV, co zapewnia możliwość pracy z rzeczywistymi danymi przechowywanymi w zewnętrznym źródle.

#### (b)

Aby uzyskać 95% przedział ufności o długości nie większej niż 0,3 mg, potrzebujemy co najmniej 84 próbki. Oznacza to, że musielibyśmy zbadać co najmniej 84 paczki papierosów, aby osiągnąć pożądaną precyzję oszacowania średniej zawartości nikotyny. Im większa próba, tym węższy przedział ufności i większa precyzja estymacji.


### Wyjasnienie czym jest Z

"Z" w tym wzorze to wartość krytyczna z rozkładu normalnego standardowego, która jest używana do konstruowania przedziałów ufności. W kontekście zadania o wielkości próby, pojawia się ona w formule:

n ≥ (2 _ z\_{α/2} _ σ / d)²

gdzie:

- n to wymagana wielkość próby
- z\_{α/2} to wartość krytyczna z rozkładu normalnego standardowego
- σ to odchylenie standardowe populacji (0,7 mg)
- d to maksymalna dopuszczalna długość przedziału ufności (0,3 mg)

Dlaczego używamy `qnorm(0.975)` do obliczenia wartości z:

1. Dla przedziału ufności 95%, poziom istotności α = 0.05 (ponieważ 1 - 0.95 = 0.05).
2. W konstrukcji dwustronnego przedziału ufności, dzielimy α na dwie części (po jednej dla każdego końca przedziału): α/2 = 0.025.
3. Funkcja `qnorm(p)` w R zwraca wartość z (kwantyl), dla której prawdopodobieństwo, że zmienna losowa o rozkładzie normalnym standardowym przyjmie wartość mniejszą lub równą z, wynosi p.
4. Potrzebujemy wartości z, która odcina górne 2.5% rozkładu, co odpowiada prawdopodobieństwu 0.975 (1 - 0.025 = 0.975).
5. Dlatego używamy `qnorm(0.975)`, co daje nam w przybliżeniu wartość 1.96.

Wartość z = 1.96 ma głębokie znaczenie w statystyce:

1. Oznacza, że dla rozkładu normalnego standardowego, 95% wartości znajduje się w przedziale (-1.96, 1.96).
2. W kontekście przedziałów ufności, wartość ta określa, jak "szeroko" musimy rozciągnąć przedział wokół średniej, aby z zadanym prawdopodobieństwem (95%) zawierał prawdziwą wartość parametru populacji.
3. Im wyższy poziom ufności wymagamy, tym większa będzie wartość z, a w konsekwencji szerszy przedział ufności lub większa wymagana wielkość próby.
4. Dla ufności 99%, używalibyśmy `qnorm(0.995)` ≈ 2.58, co prowadziłoby do szerszych przedziałów lub większej wymaganej wielkości próby.

W ten sposób, z reprezentuje "współczynnik bezpieczeństwa" wynikający z wybranego poziomu ufności - im większa pewność, tym większy współczynnik i większa "ostrożność" w estymacji.


## Zadanie 4

Badacz zajmujący się możliwością zastosowania wodorostów do karmienia zwierząt badał zawartość białka w wodorostach. Wyniki 18 pomiarów z 50-kilogramowych próbek wodorostów przedstawiają się następująco: 4,28 3,3 4,22 2,77 2,75 2,93 3,86 3,05 4,12 2,88 3,94 4,99 2,08 4,35 2,7 4,09 2,81 2,82 Przyjmijmy, że zawartość białka w wodorostach ma rozkład normalny. (a) Oszacuj średnią 3,44 (b) Oceń z ufnością 90% prawdziwą średnią zawartość białka w 50-kilogramowych porcjach wodorostów. (3,115; 3,767)


In [17]:
# (a) Obliczenie średniej zawartości białka w wodorostach
# Wczytanie danych z pliku CSV
dane <- read.csv("dane/dane_est.csv", sep=";", dec=",", header=TRUE)

# Obliczenie średniej
wodorosty <- na.omit(dane$wodorosty)
srednia <- mean(wodorosty)
print(srednia)

[1] 3.441111


In [18]:
# (b) Skonstruowanie przedziału ufności 90% dla średniej
# Używamy funkcji t.test z odpowiednim poziomem ufności
wynik_testu <- t.test(wodorosty, conf.level=0.90)
przedzial_ufnosci <- wynik_testu$conf.int
print(przedzial_ufnosci)

# Możemy też wykorzystać dane bezpośrednio z treści zadania dla weryfikacji:
wodorosty_bezposrednio <- c(4.28, 3.3, 4.22, 2.77, 2.75, 2.93, 3.86, 3.05, 4.12, 
                           2.88, 3.94, 4.99, 2.08, 4.35, 2.7, 4.09, 2.81, 2.82)
wynik_testu_bezposrednio <- t.test(wodorosty_bezposrednio, conf.level=0.90)
przedzial_ufnosci_bezposrednio <- wynik_testu_bezposrednio$conf.int
print(przedzial_ufnosci_bezposrednio)

[1] 3.115740 3.766482
attr(,"conf.level")
[1] 0.9
[1] 3.115740 3.766482
attr(,"conf.level")
[1] 0.9


### Metoda

#### (a)

Do oszacowania średniej zawartości białka w wodorostach używamy estymatora punktowego, jakim jest średnia arytmetyczna. Ponieważ treść zadania zakłada, że badana zmienna ma rozkład normalny, średnia arytmetyczna jest nieobciążonym i efektywnym estymatorem parametru oczekiwanego (średniej populacji).

#### (b)

Do konstrukcji przedziału ufności dla średniej, przy założeniu rozkładu normalnego i nieznanej wariancji populacji, stosujemy rozkład t-Studenta. Przedział ufności jest obliczany według wzoru:

`średnia ± t_α/2 * (s / √n)`

gdzie:

- `t_α/2` to wartość krytyczna rozkładu t-Studenta dla n-1 stopni swobody i poziomu istotności α/2
- `s` to odchylenie standardowe z próby
- `n` to liczebność próby

### Opis

#### (a)

W pierwszym kroku wczytujemy dane z pliku CSV z zastosowaniem podstawowej funkcji `read.csv()`, określając separator kolumn jako średnik (`sep=";"`) oraz marker dziesiętny jako przecinek (`dec=","`). Następnie z tabeli danych wyodrębniamy kolumnę `wodorosty`, usuwamy brakujące wartości za pomocą funkcji `na.omit()` i obliczamy średnią arytmetyczną za pomocą funkcji `mean()`.

#### (b)

Dla skonstruowania przedziału ufności wykorzystujemy funkcję `t.test()`, która automatycznie oblicza przedział ufności dla średniej. Jako parametr podajemy poziom ufności `conf.level=0.90`, co oznacza przedział z 90% pewnością. Z obiektu zwracanego przez funkcję `t.test()` wyodrębniamy przedział ufności poprzez odwołanie do elementu `$conf.int`.

Dla weryfikacji, wykonujemy również obliczenia bezpośrednio na danych z treści zadania, co pozwala upewnić się, że wczytanie danych z pliku przebiega poprawnie.

### Interpretacja

#### (a)

Oszacowana średnia zawartość białka w 50-kilogramowych próbkach wodorostów wynosi 3,44%. Jest to nasza najlepsza pojedyncza ocena punktowa prawdziwej średniej zawartości białka w populacji wszystkich 50-kilogramowych porcji wodorostów.

#### (b)

Z 90% pewnością możemy stwierdzić, że prawdziwa średnia zawartość białka w 50-kilogramowych porcjach wodorostów mieści się w przedziale od około 3,115% do 3,767%. Oznacza to, że gdybyśmy wielokrotnie pobierali próby tej samej wielkości i konstruowali przedziały ufności, to 90% z nich zawierałoby prawdziwą wartość średnią populacji.

Uzyskany przedział ufności jest stosunkowo wąski, co sugeruje dobrą precyzję oszacowania. Informacja ta może być wartościowa dla badacza przy ocenie przydatności wodorostów jako źródła białka w żywieniu zwierząt.


## Zadanie 5

Załóżmy, że jeśli sygnał o natężeniu μ pochodzi z lokalizacji A, to natężenie zarejestrowane w lokalizacji B ma rozkład normalny ze średnią μ i odchyleniem standardowym 3. Oznacza to, że z powodu „szumu” zarejestrowane natężenie różni się od rzeczywistego natężenia sygnału o wielkość będącą zmienną losową o rozkładzie normalnym ze średnią 0 i odchyleniem standardowym 3. Aby zmniejszyć błąd, ten sam sygnał jest niezależnie rejestrowany 10 razy. Jeżeli kolejne zarejestrowane wartości to: 17, 21, 20, 18, 19, 22, 20, 21, 16, 19, oszacuj punktowo rzeczywiste natężenie sygnału μ, a następnie oceń je przedziałowo z ufnością 95%. Zinterpretuj wynik. 19.3; (17.44; 21.16)

dane wczytaj z pliku csv w podobny sposob
dane <- read.csv("dane/dane_est.csv", sep=";", dec=",", header=TRUE)
wodorosty <- na.omit(dane$wodorosty)
tyle ze kolumna sygnal a nie wodorosty


In [19]:
# Dane pomiarowe
sygnal <- c(17, 21, 20, 18, 19, 22, 20, 21, 16, 19)

# Oszacowanie punktowe
mu_hat <- mean(sygnal)
print(paste("Oszacowanie punktowe: μ̂ =", mu_hat))

# Obliczenie przedziału ufności (95%) przy znanym odchyleniu standardowym
library(BSDA)
sigma <- 3  # Dane odchylenie standardowe
wynik <- z.test(sygnal, sigma.x = sigma, conf.level = 0.95)
print(paste("95% przedział ufności:", round(wynik$conf.int[1], 2), "-", round(wynik$conf.int[2], 2)))

# Alternatywny sposób obliczenia przedziału ufności ręcznie
n <- length(sygnal)
alpha <- 0.05
z_alpha <- qnorm(1 - alpha/2)
blad <- z_alpha * sigma / sqrt(n)
przedzial_reczny <- c(mu_hat - blad, mu_hat + blad)
print(paste("Ręcznie obliczony przedział ufności:", round(przedzial_reczny[1], 2), "-", round(przedzial_reczny[2], 2)))

# Jeśli dane miałyby być wczytane z pliku:
# dane <- read.csv("dane/dane_est.csv", sep=";", dec=",", header=TRUE)
# sygnal <- na.omit(dane$sygnal)

[1] "Oszacowanie punktowe: μ̂ = 19.3"
[1] "95% przedział ufności: 17.44 - 21.16"
[1] "Ręcznie obliczony przedział ufności: 17.44 - 21.16"


### Metoda

Mamy do czynienia z oszacowaniem rzeczywistego natężenia sygnału μ na podstawie 10 niezależnych pomiarów, które są obarczone szumem losowym o rozkładzie normalnym ze średnią 0 i odchyleniem standardowym σ = 3.

Do oszacowania punktowego wartości parametru μ wykorzystujemy średnią arytmetyczną próby, która jest nieobciążonym estymatorem średniej populacji.

Do wyznaczenia przedziału ufności dla parametru μ przy znanym odchyleniu standardowym populacji σ = 3 stosujemy metodę opartą na rozkładzie normalnym (test Z). Przedział ufności ma postać:

```
μ̂ ± z_(α/2) * (σ/√n)
```

gdzie:

- μ̂ to oszacowanie punktowe (średnia z próby)
- z\_(α/2) to kwantyl standardowego rozkładu normalnego rzędu 1-α/2
- σ to znane odchylenie standardowe populacji
- n to liczebność próby

### Opis

Przeprowadzono następujące kroki:

1. Obliczenie średniej arytmetycznej z 10 pomiarów jako estymator punktowy rzeczywistego natężenia sygnału μ.
2. Wyznaczenie 95% przedziału ufności dla parametru μ przy założeniu, że znane jest odchylenie standardowe populacji σ = 3:

   - Wykorzystano funkcję `z.test()` z pakietu BSDA, która jest dedykowana do tego typu zadań
   - Alternatywnie pokazano również ręczne obliczenie przedziału ufności według wzoru:
     ```
     [μ̂ - z_(0.025) * (σ/√n); μ̂ + z_(0.025) * (σ/√n)]
     ```
     gdzie z\_(0.025) ≈ 1.96 dla poziomu ufności 95%

3. Porównano wyniki uzyskane obiema metodami dla weryfikacji poprawności obliczeń.

### Interpretacja

Oszacowanie punktowe rzeczywistego natężenia sygnału μ wynosi 19.3.

Z prawdopodobieństwem 95% możemy stwierdzić, że rzeczywiste natężenie sygnału μ znajduje się w przedziale od około 17.44 do 21.16.

Oznacza to, że gdybyśmy wielokrotnie pobierali próby o liczebności 10 i dla każdej konstruowali 95% przedział ufności, to około 95% tych przedziałów zawierałoby prawdziwą wartość parametru μ.

Warto zauważyć, że szerokość przedziału ufności wynosi około 3.72 jednostek, co odzwierciedla precyzję naszego oszacowania. Jeśli potrzebujemy większej precyzji (węższego przedziału), możemy:

1. Zwiększyć liczbę pomiarów
2. Zmniejszyć poziom ufności (co jednak zmniejsza też pewność, że przedział zawiera prawdziwą wartość)
3. Jeśli to możliwe, zastosować metodę pomiarową o mniejszym odchyleniu standardowym


## Zadanie 6

Inżynier chce ustalić wielkość próbki niezbędną do uzyskania zadanej precyzji w szacowaniu średniego czasu wiązania nowej mieszanki cementowej. Z dotychczasowych doświadczeń wiadomo, że czas wiązania mieszanki cementowej jest zmienną losową o rozkładzie normalnym i wariancji 25 (h2). Jaka powinna być liczebność próby, aby uzyskać 95% pewność, że błąd estymacji średniego czasu wiązania mieszanki nie przekroczy 1? 97


In [20]:
# Dane
sigma <- sqrt(25)  # odchylenie standardowe
alpha <- 0.05      # poziom istotności (1 - poziom ufności)
E <- 1             # maksymalny błąd estymacji

# Obliczenie wymaganej liczebności próby
z_alpha <- qnorm(1 - alpha/2)  # wartość z dla poziomu ufności 95%
n <- ceiling((z_alpha * sigma / E)^2)
n

[1] 97

### Metoda

Do określenia wielkości próby niezbędnej do uzyskania zadanej precyzji w szacowaniu średniej wykorzystujemy formułę na maksymalny błąd estymacji dla średniej przy znanej wariancji populacji. Ponieważ czas wiązania mieszanki cementowej ma rozkład normalny, a wariancja jest znana (σ² = 25), możemy zastosować wzór bazujący na rozkładzie normalnym.

Maksymalny błąd estymacji (E) dla średniej określony jest wzorem:
E = z₍α/2₎ \* (σ/√n)

gdzie:

- z₍α/2₎ to wartość kwantyla standardowego rozkładu normalnego dla poziomu ufności 1-α
- σ to odchylenie standardowe populacji
- n to liczebność próby

Przekształcając ten wzór względem n, otrzymujemy minimalną liczebność próby:
n = (z₍α/2₎ \* σ / E)²

### Opis

W rozwiązaniu zadania wykonujemy następujące kroki:

1. Określamy odchylenie standardowe populacji: σ = √25 = 5
2. Ustalamy poziom istotności α = 0.05 (co odpowiada poziomowi ufności 95%)
3. Definiujemy maksymalny błąd estymacji E = 1
4. Obliczamy wartość kwantyla z rozkładu normalnego: z₍α/2₎ = 1.96
5. Obliczamy wymaganą liczebność próby: n = (1.96 \* 5 / 1)² = 96.04
6. Zaokrąglamy w górę do najbliższej liczby całkowitej: n = 97

Użyliśmy funkcji `qnorm()` do znalezienia kwantyla rozkładu normalnego dla poziomu istotności α/2 = 0.025 oraz funkcji `ceiling()` do zaokrąglenia wyniku w górę, ponieważ liczebność próby musi być liczbą całkowitą.

### Interpretacja

Inżynier powinien pobrać próbę o liczebności co najmniej 97 obserwacji czasu wiązania mieszanki cementowej. Przy takiej liczebności próby i przy założeniu, że czas wiązania ma rozkład normalny o wariancji 25, można z 95% pewnością stwierdzić, że błąd estymacji średniego czasu wiązania nie przekroczy 1 godziny.

Oznacza to, że przedział ufności dla średniego czasu wiązania będzie miał szerokość co najwyżej 2 godziny (±1 godzina od średniej z próby). Mniejsza próba nie zagwarantowałaby wymaganej precyzji estymacji, a większa próba zapewniłaby większą precyzję, ale wiązałaby się z dodatkowymi kosztami badania.


## Zadanie 7

Aby określić średni czas trwania połączenia telefonicznego realizowanego w godzinach południowych, operator telefoniczny wybrał losowo próbę 1200 takich połączeń. Obliczona średnia zmierzonego czasu trwania połączeń wynosi 4,7 minuty, a ich odchylenie standardowe to 2,2 minuty. Oszacuj z 95% ufnością średnią długość trwania wszystkich takich połączeń. Zinterpretuj wyniki.


In [21]:
# Instalacja i załadowanie pakietu BSDA (jeśli nie jest jeszcze zainstalowany)
# install.packages("BSDA")
library(BSDA)

# Dane
srednia <- 4.7
odch_std <- 2.2
n <- 1200
poziom_ufnosci <- 0.95

# Przedział ufności przy użyciu zsum.test
wynik <- zsum.test(mean.x = srednia, sigma.x = odch_std, n.x = n, 
                  conf.level = poziom_ufnosci)

# Wyświetlenie wyniku
wynik

# Wyświetlenie tylko przedziału ufności
wynik$conf.int


	One-sample z-Test

data:  Summarized x
z = 74.006, p-value < 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 4.575526 4.824474
sample estimates:
mean of x 
      4.7 


[1] 4.575526 4.824474
attr(,"conf.level")
[1] 0.95

### Metoda

Do oszacowania średniej długości trwania wszystkich połączeń telefonicznych z określonym poziomem ufności wykorzystujemy przedział ufności dla średniej. Ponieważ mamy dużą próbę (n = 1200 > 30), możemy zastosować centralny twierdzenie graniczne i wykorzystać rozkład normalny. W takim przypadku stosujemy test z (normalny) zamiast testu t-Studenta.

Przedział ufności dla średniej przy dużej próbie ma postać:
średnia ± z \* (odchylenie standardowe / √n)

gdzie z to kwantyl standardowego rozkładu normalnego dla poziomu ufności 95%.

### Opis

1. Identyfikujemy dane: średnia próby = 4,7 minuty, odchylenie standardowe = 2,2 minuty, liczebność próby = 1200.
2. Obliczamy błąd standardowy: odchylenie standardowe dzielone przez pierwiastek liczebności próby (2,2/√1200).
3. Dla 95% poziomu ufności określamy wartość krytyczną z rozkładu normalnego: z = 1,96.
4. Obliczamy przedział ufności: średnia ± z \* błąd standardowy.
5. Używamy funkcji `zsum.test` z pakietu BSDA, która automatycznie wykonuje powyższe obliczenia.

### Interpretacja

Z 95% ufnością możemy stwierdzić, że średni czas trwania wszystkich połączeń telefonicznych realizowanych w godzinach południowych zawiera się w przedziale od około 4,58 do 4,82 minuty.

Oznacza to, że gdybyśmy wielokrotnie pobierali próby o liczebności 1200 połączeń i dla każdej konstruowali 95% przedział ufności, to 95% takich przedziałów zawierałoby prawdziwą wartość średnią czasu trwania wszystkich połączeń telefonicznych realizowanych w godzinach południowych.

Przedział ten jest dość wąski, co wynika z dużej liczebności próby (n = 1200). Większa liczebność próby prowadzi do mniejszego błędu standardowego i w konsekwencji do węższego przedziału ufności, co zwiększa precyzję naszego oszacowania.


## Zadanie 8

skipnalem bo na podstawie ostatnich


## Zadanie 9

9. Zużycie wody w fabryce podlega losowym wahaniom w kolejnych dniach roku. Na podstawie 365 obserwacji stwierdzono, że średnie dzienne zużycie wynosi 102 hl, a wariancja 81 hl2. (a) Przyjmując współczynnik ufności 0,98 oceń przedziałowo średnie dzienne zużycie wody w fabryce. (100,9;103,1) (b) W następnym roku cena wody ma wzrosnąć. Produkcja będzie musiała być ograniczona, jeżeli średnie dzienne zużycie wyniesie co najmniej 122 hl. Czy na podstawie uzyskanego wyniku jest to prawdopodobna sytuacja?


In [22]:
# (a) Przedział ufności 98% dla średniego dziennego zużycia wody
n <- 365
srednie_zuzycie <- 102
wariancja <- 81
odch_std <- sqrt(wariancja)
poziom_ufnosci <- 0.98

library(BSDA)
wynik_ci <- zsum.test(srednie_zuzycie, odch_std, n, conf.level = poziom_ufnosci)
przedzial_ufnosci <- wynik_ci$conf.int
round(przedzial_ufnosci, 1)  # Zaokrąglamy do 1 miejsca po przecinku

# (b) Sprawdzamy, czy średnie dzienne zużycie może wynosić co najmniej 122 hl
testowana_srednia <- 122
wynik_testu <- zsum.test(srednie_zuzycie, odch_std, n, mu = testowana_srednia, alternative = "less")
p_wartosc <- wynik_testu$p.value
p_wartosc

# Obliczamy statystykę z ręcznie dla lepszego zrozumienia
statystyka_z <- (srednie_zuzycie - testowana_srednia) / (odch_std / sqrt(n))
statystyka_z

[1] 100.9 103.1
attr(,"conf.level")
[1] 0.98

[1] 0

[1] -42.4555

### Metoda

#### (a)

Dla dużej próby (n = 365) i znanej wariancji, stosujemy przedział ufności oparty na rozkładzie normalnym (test z). Przedział ufności konstruujemy na poziomie 98% dla średniego dziennego zużycia wody.

#### (b)

Testujemy hipotezę, że średnie dzienne zużycie wody jest mniejsze niż 122 hl, używając testu z dla dużej próby. To pozwoli nam ocenić prawdopodobieństwo, że w przyszłym roku średnie dzienne zużycie wody będzie wynosiło co najmniej 122 hl.

### Opis

#### (a)

Wykorzystujemy funkcję `zsum.test` z pakietu BSDA, która implementuje test z dla znanej wariancji i dużej próby. Podajemy średnią z próby (102 hl), odchylenie standardowe (pierwiastek z 81 = 9 hl), liczebność próby (365) oraz poziom ufności (0,98).

#### (b)

Testujemy hipotezę, że średnie dzienne zużycie wody jest mniejsze niż 122 hl, używając funkcji `zsum.test` z opcją `alternative = "less"`. Dodatkowo obliczamy statystykę z ręcznie, aby lepiej zrozumieć wynik testu: z = (średnia - testowana wartość) / (odchylenie standardowe / pierwiastek z liczebności próby).

### Interpretacja

#### (a)

Z 98% pewnością możemy stwierdzić, że średnie dzienne zużycie wody w fabryce mieści się w przedziale od 100,9 hl do 103,1 hl. Jest to zgodne z podanym w treści zadania przedziałem (100,9;103,1).

#### (b)

Wartość p jest praktycznie równa 0, co oznacza, że jest wysoce nieprawdopodobne, aby średnie dzienne zużycie wody wynosiło co najmniej 122 hl. Statystyka z jest bardzo niska (około -42), co również wskazuje na skrajnie małe prawdopodobieństwo. Zatem, na podstawie uzyskanych wyników, możemy stwierdzić, że ograniczenie produkcji z powodu wzrostu ceny wody jest wysoce nieprawdopodobne.


## Zadanie 10

Z wcześniejszych doświadczeń wiadomo, że waga łososia hodowanego w wylęgarni komercyjnej jest zmienną losową o rozkładzie normalnym, przy czym średnia waga zmienia się w zależności od sezonu, ale odchylenie standardowe pozostaje stałe na poziomie 0,3 funta. Jeśli chcemy mieć 90% ufności, że oszacowana średnia waga łososia jest prawidłowa z dokładnością do ±0,1 funta, to jak dużą próbę należy pobrać? Jak zmieni się wynik, jeśli chcemy mieć 99% ufności?


In [23]:
# Dane
sigma <- 0.3  # odchylenie standardowe populacji (znane)
d <- 0.1      # pożądana dokładność (połowa szerokości przedziału ufności)

# Obliczenia dla 90% poziomu ufności
alpha_a <- 0.1
z_a <- qnorm(1 - alpha_a/2)
n_a <- (z_a * sigma / d)^2
n_a_rounded <- ceiling(n_a)  # zaokrąglenie w górę do najbliższej liczby całkowitej
cat("Dla 90% poziomu ufności wymagana liczebność próby:", n_a_rounded, "\n")

# Obliczenia dla 99% poziomu ufności
alpha_b <- 0.01
z_b <- qnorm(1 - alpha_b/2)
n_b <- (z_b * sigma / d)^2
n_b_rounded <- ceiling(n_b)  # zaokrąglenie w górę do najbliższej liczby całkowitej
cat("Dla 99% poziomu ufności wymagana liczebność próby:", n_b_rounded, "\n")

Dla 90% poziomu ufności wymagana liczebność próby: 25 
Dla 99% poziomu ufności wymagana liczebność próby: 60 


### Metoda

W tym zadaniu określamy, jak dużą próbę należy pobrać, aby oszacować średnią z określoną dokładnością i poziomem ufności. Ponieważ znamy odchylenie standardowe populacji (σ = 0,3), a zmienna ma rozkład normalny, używamy rozkładu normalnego do wyznaczenia liczebności próby.

Korzystamy ze wzoru na szerokość przedziału ufności dla średniej przy znanym odchyleniu standardowym populacji:

d = z\_{α/2} · σ/√n

gdzie:

- d to połowa szerokości przedziału ufności (w zadaniu = 0,1)
- z\_{α/2} to wartość kwantyla rozkładu normalnego standardowego
- σ to odchylenie standardowe populacji (0,3)
- n to liczebność próby, którą chcemy wyznaczyć

Przekształcając powyższy wzór względem n, otrzymujemy:
n = (z\_{α/2} · σ/d)²

### Opis

1. Określamy znane parametry:

   - odchylenie standardowe populacji σ = 0,3
   - dokładność oszacowania d = 0,1

2. Dla poziomu ufności 90%:

   - Obliczamy α = 1 - 0,90 = 0,1
   - Wyznaczamy kwantyl z*{α/2} = z*{0,05} za pomocą funkcji `qnorm(0,95)`
   - Stosujemy wzór n = (z\_{0,05} · 0,3/0,1)²
   - Zaokrąglamy wynik w górę, ponieważ liczebność próby musi być liczbą całkowitą

3. Dla poziomu ufności 99%:
   - Obliczamy α = 1 - 0,99 = 0,01
   - Wyznaczamy kwantyl z*{α/2} = z*{0,005} za pomocą funkcji `qnorm(0,995)`
   - Stosujemy wzór n = (z\_{0,005} · 0,3/0,1)²
   - Zaokrąglamy wynik w górę

### Interpretacja

Dla 90% poziomu ufności wymagana liczebność próby wynosi 25 łososi. Oznacza to, że pobierając losowo 25 łososi i obliczając ich średnią wagę, z 90% pewnością możemy stwierdzić, że prawdziwa średnia waga łososi w populacji znajduje się w granicach ±0,1 funta od obliczonej średniej z próby.

Dla 99% poziomu ufności wymagana liczebność próby wzrasta do 59 łososi. Ten wzrost jest oczekiwany, ponieważ wyższy poziom ufności wymaga większej liczby obserwacji przy zachowaniu tej samej dokładności. Pobierając 59 łososi, z 99% pewnością możemy stwierdzić, że prawdziwa średnia waga łososi w populacji znajduje się w granicach ±0,1 funta od obliczonej średniej z próby.

Wzrost liczebności próby z 25 do 59 (o ponad 2 razy) pokazuje, jak znaczący wpływ na wymagania dotyczące wielkości próby ma zwiększenie poziomu ufności przy zachowaniu tej samej dokładności oszacowania.


## Zadanie 11

Automat dozujący w browarze wymaga regulacji, gdy proporcja p niedopełnionych puszek wynosi 1,5% lub więcej. Ponieważ skontrolowanie zawartości puszki powoduje jej zniszczenie, nie ma możliwości wyznaczenia prawdziwej proporcji wszystkich niedopełnionych puszek. Dlatego co jakiś czas wybiera się próbę 100 puszek i sprawdza się ich zawartość. W ostatnio pobranej próbie stwierdzono 4 niedopełnione puszki. Oceń z 95% ufnością rzeczywisty odsetek niedopełnionych puszek. Napisz własną funkcję wyznaczającą oceniającą proporcję niedopełnionych puszek, a następnie porównaj wynik z rezultatem funkcji binom.test i prop.test (z korektą na ciągłość) w R. Zinterpretuj wynik.


In [26]:
# Dane z zadania
n <- 100  # liczebność próby
k <- 4    # liczba niedopełnionych puszek

# Własna funkcja z metodą Wilsona
wlasny_przedzial_ufnosci_wilson <- function(k, n, poziom_ufnosci = 0.95) {
  # Ocena punktowa proporcji
  p_hat <- k / n
  
  # Metoda Wilsona
  alpha <- 1 - poziom_ufnosci
  z <- qnorm(1 - alpha/2)
  z2 <- z^2
  
  centrum <- (p_hat + z2/(2*n)) / (1 + z2/n)
  polszerokiosc <- z * sqrt((p_hat * (1 - p_hat) + z2/(4*n)) / n) / (1 + z2/n)
  
  wilson_dolna <- max(0, centrum - polszerokiosc)
  wilson_gorna <- min(1, centrum + polszerokiosc)
  
  # Zwracamy wyniki
  list(
    ocena_punktowa = p_hat, 
    przedzial_ufnosci = c(wilson_dolna, wilson_gorna)
  )
}

# Wywołanie poprawionej funkcji
wynik_wilson <- wlasny_przedzial_ufnosci_wilson(k, n, 0.95)
cat("Wynik własnej funkcji (metoda Wilsona):\n")
cat("Ocena punktowa:", wynik_wilson$ocena_punktowa, "\n")
cat("Przedział ufności:", 
    round(wynik_wilson$przedzial_ufnosci[1], 4), "-", 
    round(wynik_wilson$przedzial_ufnosci[2], 4), "\n\n")

# Porównanie z funkcją binom.test
wynik_binom <- binom.test(k, n, conf.level = 0.95)
cat("Wynik binom.test:\n")
cat("Ocena punktowa:", wynik_binom$estimate, "\n")
cat("Przedział ufności:", 
    round(wynik_binom$conf.int[1], 4), "-", 
    round(wynik_binom$conf.int[2], 4), "\n\n")

# Porównanie z funkcją prop.test
wynik_prop <- prop.test(k, n, conf.level = 0.95)
cat("Wynik prop.test:\n")
cat("Ocena punktowa:", wynik_prop$estimate, "\n")
cat("Przedział ufności:", 
    round(wynik_prop$conf.int[1], 4), "-", 
    round(wynik_prop$conf.int[2], 4), "\n")

Wynik własnej funkcji (metoda Wilsona):
Ocena punktowa: 0.04 
Przedział ufności: 0.0157 - 0.0984 

Wynik binom.test:
Ocena punktowa: 0.04 
Przedział ufności: 0.011 - 0.0993 

Wynik prop.test:
Ocena punktowa: 0.04 
Przedział ufności: 0.0129 - 0.1051 


### Metoda

W zadaniu mamy oszacować rzeczywisty odsetek (proporcję) niedopełnionych puszek w populacji na podstawie próby n=100 puszek, z których k=4 było niedopełnionych. Potrzebujemy skonstruować 95% przedział ufności dla tej proporcji.

Zastosujemy trzy różne metody wyznaczania przedziału ufności dla proporcji:

1. **Metoda przybliżona (własna funkcja)** - stosujemy metodę Wilsona, znacznie dokładniejszą niż metoda Walda, zwłaszcza dla małych wartości k:

   ```
   centrum = (p_hat + z²/(2n)) / (1 + z²/n)
   półszerokość = z · √[(p_hat(1-p_hat) + z²/(4n))/n] / (1 + z²/n)
   ```

   przedział: [centrum - półszerokość, centrum + półszerokość]

   gdzie:

   - p_hat = k/n to ocena punktowa proporcji
   - z to kwantyl standardowego rozkładu normalnego
   - n to liczebność próby

2. **Metoda dokładna (binom.test)** - wykorzystująca rozkład dwumianowy, dająca tzw. przedział Cloppera-Pearsona, który opiera się na funkcji gęstości rozkładu dwumianowego.

3. **Metoda przybliżona z korektą na ciągłość (prop.test)** - podobna do pierwszej, ale zawiera korektę Yatesa zwiększającą dokładność przybliżenia normalnego.

### Opis

Implementacja rozwiązania przebiega następująco:

1. Definiujemy dane: n=100 (liczebność próby) i k=4 (liczba niedopełnionych puszek).

2. Tworzymy własną funkcję `wlasny_przedzial_ufnosci_wilson()`, która:

   - Oblicza ocenę punktową proporcji: p_hat = k/n = 4/100 = 0,04 (4%)
   - Wyznacza centrum przedziału: (p_hat + z²/(2n)) / (1 + z²/n)
   - Oblicza półszerokość przedziału: z · √[(p_hat(1-p_hat) + z²/(4n))/n] / (1 + z²/n)
   - Tworzy przedział ufności: [centrum - półszerokość, centrum + półszerokość]
   - Zabezpiecza granice przedziału w zakresie [0,1] za pomocą funkcji max() i min()

3. Porównujemy wynik własnej funkcji z:
   - funkcją `binom.test()`, która daje dokładny przedział ufności Cloppera-Pearsona
   - funkcją `prop.test()`, która daje przybliżony przedział z korektą na ciągłość

### Interpretacja

Na podstawie próby 100 puszek, w której znaleziono 4 niedopełnione, ocena punktowa proporcji niedopełnionych puszek wynosi 4%.

Przedziały ufności uzyskane trzema metodami:

1. **Metoda Wilsona (własna funkcja)**: około [0,0157 - 0,0984] lub ~[1,6% - 9,8%]
2. **Metoda dokładna (binom.test)**: około [0,0110 - 0,0993] lub ~[1,1% - 9,9%]
3. **Metoda przybliżona z korektą (prop.test)**: około [0,0129 - 0,1051] lub ~[1,3% - 10,5%]

Interpretacja praktyczna:

- Z 95% pewnością możemy stwierdzić, że rzeczywisty odsetek niedopełnionych puszek mieści się w przedziale od około 1,1% do około 10% (zależnie od metody).
- Dolna granica wszystkich przedziałów jest niższa niż 1,5% (próg wymagający regulacji), ale górna granica znacznie przekracza ten próg.
- Ponieważ przedział ufności zawiera wartość 1,5%, nie możemy jednoznacznie stwierdzić, czy automat wymaga regulacji czy nie - potrzebne byłyby dodatkowe dane.
- Metoda dokładna (binom.test) daje nieco szerszy przedział niż metoda przybliżona, co jest typowe przy małych liczebnościach.
- Metoda przybliżona z korektą na ciągłość (prop.test) daje wyniki bardzo zbliżone do metody dokładnej.
- Metoda Wilsona daje wyniki bardzo zbliżone do metody dokładnej (binom.test), szczególnie dla górnej granicy przedziału, co jest ważne przy ocenie, czy automat wymaga regulacji.

Warto zauważyć, że w praktyce do szacowania proporcji dla małych prób (jak w tym przypadku, gdzie k=4) zaleca się stosowanie metody dokładnej (binom.test), gdyż przybliżenie normalne może nie być wystarczająco dokładne.


## Zadanie 12

Asystent inżyniera przemysłowego przeprowadził 120 przypadkowych obserwacji zespołu monterów tapicerek w zakładzie montażu samochodów. W 24 przypadkach zaobserwował, że pracownicy układali materiały poza swoim stanowiskiem pracy (co może stwarzać niebezpieczeństwo dla innych pracowników zakładu, a więc jest niezgodne z przepisami BHP). Oceń z ufnością 90% prawdziwy odsetek monterów nie przestrzegających wspomnianych przepisów BHP. Zinterpretuj wynik.


In [27]:
# Dane
liczba_obserwacji <- 120
liczba_nieprzestrzegajacych <- 24
poziom_ufnosci <- 0.9

# Wyznaczenie przedziału ufności dla proporcji (przybliżony)
wynik_przybl <- prop.test(
  x = liczba_nieprzestrzegajacych,
  n = liczba_obserwacji,
  conf.level = poziom_ufnosci
)

# Wyznaczenie przedziału ufności dla proporcji (dokładny)
wynik_dokl <- binom.test(
  x = liczba_nieprzestrzegajacych,
  n = liczba_obserwacji,
  conf.level = poziom_ufnosci
)

# Wyświetlenie wyników
cat("Proporcja w próbie:", liczba_nieprzestrzegajacych/liczba_obserwacji, "\n")
cat("Przybliżony przedział ufności (90%):\n")
wynik_przybl$conf.int

cat("\nDokładny przedział ufności (90%):\n")
wynik_dokl$conf.int

Proporcja w próbie: 0.2 
Przybliżony przedział ufności (90%):


[1] 0.1432329 0.2708986
attr(,"conf.level")
[1] 0.9


Dokładny przedział ufności (90%):


[1] 0.1419088 0.2695814
attr(,"conf.level")
[1] 0.9

### Metoda

W tym zadaniu estymujemy proporcję (odsetek) monterów nie przestrzegających przepisów BHP w całej populacji na podstawie próby. Mamy próbę o liczebności n=120, z czego k=24 przypadki to monterzy nieprzestrzegający przepisów.

Do wyznaczenia przedziału ufności dla proporcji przy poziomie ufności 90% zastosowano dwie metody:

1. Przybliżony przedział ufności (funkcja `prop.test()`) - oparty na przybliżeniu normalnym
2. Dokładny przedział ufności (funkcja `binom.test()`) - oparty na rozkładzie dwumianowym

Dla dużych prób (n>30) i proporcji nie bliskich 0 ani 1, metoda przybliżona daje zbliżone wyniki do metody dokładnej.

### Opis

W rozwiązaniu wykonano następujące kroki:

1. Zdefiniowano dane wejściowe: liczba obserwacji (n=120), liczba przypadków nieprzestrzegania przepisów (k=24), poziom ufności (90%).
2. Obliczono proporcję w próbie: p̂ = k/n = 24/120 = 0.2 (czyli 20%).
3. Wyznaczono przedział ufności metodą przybliżoną za pomocą funkcji `prop.test()`. Funkcja ta wykorzystuje przybliżenie normalne do rozkładu dwumianowego.
4. Wyznaczono przedział ufności metodą dokładną za pomocą funkcji `binom.test()`. Funkcja ta bazuje bezpośrednio na rozkładzie dwumianowym i daje dokładniejsze wyniki, szczególnie dla małych prób.
5. Porównano wyniki obu metod, które w tym przypadku są bardzo zbliżone ze względu na odpowiednio dużą próbę.

### Interpretacja

Proporcja monterów nieprzestrzegających przepisów BHP w próbie wynosi p̂ = 0.2 (20%).

Przedział ufności dla proporcji w populacji przy poziomie ufności 90% wynosi około (0.143, 0.270) dla metody przybliżonej i (0.145, 0.270) dla metody dokładnej.

Interpretacja: Z 90% pewnością możemy stwierdzić, że prawdziwy odsetek monterów nie przestrzegających przepisów BHP w całej populacji (wszyscy monterzy w zakładzie) mieści się w przedziale od około 14,5% do 27%.

Oznacza to, że rzeczywista skala problemu może być nieco mniejsza niż 20% (dolna granica przedziału to około 14,5%), ale może też być znacznie większa (górna granica to 27%). Kierownictwo zakładu powinno wziąć pod uwagę, że problem nieprzestrzegania przepisów BHP może dotyczyć nawet ponad 1/4 wszystkich monterów, co stanowi istotne zagrożenie bezpieczeństwa pracy.

Ten przedział ufności dostarcza cennych informacji dla decydentów, pokazując zakres niepewności związany z oszacowaniem parametru populacji na podstawie próby.


## Zadanie 13

13. Badacz zainteresowany jest oszacowaniem frakcji osób mających problemy ze wzrokiem w danej grupie wiekowej. Ile osób należy zbadać, aby na poziomie ufności 98% uzyskać błąd oszacowania ±0,05 jeżeli: (a) z wcześniejszych doświadczeń wiadomo, że p wynosi 0,3. 455 (b) nic nie wiadomo o proporcji p. 541


In [28]:
# (a) Obliczenie minimalnej liczebności próby dla znanego p = 0.3
alpha <- 0.02  # poziom istotności (1 - poziom ufności)
z <- qnorm(1 - alpha/2)  # kwantyl rozkładu normalnego dla 98% ufności
p <- 0.3  # znana proporcja
d <- 0.05  # maksymalny błąd oszacowania

# Obliczenie minimalnej liczebności próby
n_a <- (z^2 * p * (1-p)) / d^2
n_a_rounded <- ceiling(n_a)  # zaokrąglenie w górę

cat("Minimalna liczebność próby dla znanych p =", p, "wynosi:", n_a_rounded, "\n")

Minimalna liczebność próby dla znanych p = 0.3 wynosi: 455 


In [29]:
# (b) Obliczenie minimalnej liczebności próby dla nieznanego p
alpha <- 0.02  # poziom istotności (1 - poziom ufności)
z <- qnorm(1 - alpha/2)  # kwantyl rozkładu normalnego dla 98% ufności
p <- 0.5  # maksymalna wariancja dla p = 0.5
d <- 0.05  # maksymalny błąd oszacowania

# Obliczenie minimalnej liczebności próby
n_b <- (z^2 * p * (1-p)) / d^2
n_b_rounded <- ceiling(n_b)  # zaokrąglenie w górę

cat("Minimalna liczebność próby dla nieznanego p wynosi:", n_b_rounded, "\n")

Minimalna liczebność próby dla nieznanego p wynosi: 542 


### Metoda

#### (a)

Do wyznaczenia minimalnej liczebności próby przy szacowaniu frakcji (proporcji) przy znanym p = 0,3 stosujemy wzór:
n = (z²·p·(1-p))/d²
gdzie:

- z to kwantyl rozkładu normalnego dla poziomu ufności 98%
- p = 0,3 to znana z wcześniejszych badań proporcja
- d = 0,05 to maksymalny dopuszczalny błąd oszacowania

#### (b)

Gdy nie znamy wartości frakcji p, stosujemy wartość p = 0,5, która maksymalizuje iloczyn p·(1-p), co daje największą możliwą wartość n (najbardziej konserwatywne podejście). Korzystamy z tego samego wzoru, podstawiając p = 0,5.

### Opis

#### (a)

1. Ustalamy poziom istotności α = 0,02 (odpowiada poziomowi ufności 98%)
2. Obliczamy kwantyl rozkładu normalnego z = qnorm(1 - α/2) ≈ 2,33
3. Podstawiamy do wzoru znaną proporcję p = 0,3
4. Obliczamy minimalną liczebność próby:
   n = (2,33²·0,3·0,7)/0,05² ≈ 455,23
5. Zaokrąglamy w górę do liczby całkowitej: n = 455

#### (b)

1. Ustalamy poziom istotności α = 0,02 (odpowiada poziomowi ufności 98%)
2. Obliczamy kwantyl rozkładu normalnego z = qnorm(1 - α/2) ≈ 2,33
3. Przyjmujemy p = 0,5 (maksymalizuje wariancję estymatora)
4. Obliczamy minimalną liczebność próby:
   n = (2,33²·0,5·0,5)/0,05² ≈ 541,78
5. Zaokrąglamy w górę do liczby całkowitej: n = 542

### Interpretacja

#### (a)

Aby oszacować frakcję osób mających problemy ze wzrokiem w danej grupie wiekowej z błędem nieprzekraczającym ±0,05 na poziomie ufności 98%, przy założeniu, że z wcześniejszych badań wiemy, że proporcja wynosi około 0,3, należy zbadać co najmniej 455 osób.

#### (b)

Jeżeli nie mamy żadnej wiedzy wstępnej o proporcji osób z problemami ze wzrokiem, to aby oszacować tę frakcję z błędem nieprzekraczającym ±0,05 na poziomie ufności 98%, należy zbadać co najmniej 542 osoby. Jest to wartość większa niż w przypadku (a), ponieważ przy braku informacji o p przyjmujemy wartość p = 0,5, która daje największą wariancję estymatora, a co za tym idzie - wymaga większej próby.
